File: frontend/src/types/feature.ts
export interface FeatureStoreRecord {
  id: string;
  feature_vector_id: string;
  transaction_id: string;
  merchant_id?: string;
  customer_id?: string;
  device_id?: string;
  feature_version: string;
  feature_group: string;
  feature_count: number;
  feature_payload: Record<string, any>;
  prediction_ready: boolean;
  created_at: string;
  updated_at: string;
}

export interface PaginatedFeatureStore {
  items: FeatureStoreRecord[];
  total: number;
  page: number;
  size: number;
  pages: number;
}

export interface FeatureFilterParams {
  transaction_id?: string;
  feature_version?: string;
  prediction_ready?: boolean;
  page?: number;
  size?: number;
  sort_by?: string;
  sort_dir?: string;
}

File: backend/app/schemas/feature.py
import uuid
from datetime import datetime
from typing import Any, Dict, List, Optional
from pydantic import BaseModel, ConfigDict, Field


class FeatureGenerateRequest(BaseModel):
    transaction_id: str = Field(..., example="TXN-8D93240D", description="Target Transaction ID")


class FeatureRecomputeRequest(BaseModel):
    transaction_id: str = Field(..., example="TXN-8D93240D")


class FeatureStoreResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    feature_vector_id: str
    transaction_id: str
    merchant_id: Optional[uuid.UUID] = None
    customer_id: Optional[str] = None
    device_id: Optional[str] = None
    feature_version: str
    feature_group: str
    feature_count: int
    feature_payload: Dict[str, Any]
    prediction_ready: bool
    created_at: datetime
    updated_at: datetime


class PaginatedFeatureStoreResponse(BaseModel):
    items: List[FeatureStoreResponse]
    total: int
    page: int
    size: int
    pages: int
